apps/avifenc.c

```diff
@@ -1561,6 +1564,15 @@ int main(int argc, char * argv[])
         goto cleanup;
     }
 
+    if (noOverwrite){
+        FILE * outfile = fopen(outputFilename, "rb");
+        if (outfile) {
+            fclose(outfile);
+            printf("No Overwrite: %s\n", outputFilename);
```

Comment:
Write the error message to stderr and use a more descriptive message, e.g.:
fprintf(stderr, "ERROR: output file %s already exists and --no-overwrite was specified\n", outputFilename));

Also I think this check should also be done just before writing the output file. Encodes can take a long time and the file might have been created in between. Maybe put the check in a helper function and call it here (before encoding, as early exit), and again before writing to be safe.

Reply:
Note: `fopen()` could be used at writing time with [file access mode flag `"x"`](https://en.cppreference.com/w/c/io/fopen#File_access_flags) instead of doing a separate `fopen("r")` beforehand but unfortunately it is a C11 feature and [libavif is only C99](https://github.com/AOMediaCodec/libavif/blob/ddffb7758872f35ef6703791d353e37decb3b7d9/CMakeLists.txt#L17). Also as you mentioned encoding can be slow so an early check is necessary anyway.